### ISSUE TITLE
there is bug in `Lint/DuplicateHashKey` cop

### ISSUE THREAD

**Issue by Ivanov-Anton** [rubocop/rubocop#11953]
The "Lint/DuplicateHashKey" cop in Robocop does not correctly flag offenses for duplicate hash keys in the provided code snippets.

### Code Snippet 1:
```
ACTIONS = {
  HAS_1 => {
    SUB_HASH_1 => 'string',
    SUB_HASH_2 => 'string',
    SUB_HASH_3 => 'string',
    SUB_HASH_4 => 'string',
    SUB_HASH_5 => 'string',
    SUB_HASH_1 => 'string'
    
  }.freeze,
}.freeze
```

### Expected Behavior:

According to the "Lint/DuplicateHashKey" cop, the duplicate key "SUB_HASH_1" should be flagged as an offense. However, in the provided code snippet, the cop does not generate any offense and ignores the duplicate key.

### Code Snippet 2:
```
resource_classes = {
  PAS::REST::MAIN => {
    plan: :main_plan
  },
  RAM::REST::MAIN => {
    second_plan: :second_plan_value
  },
  PAS::REST::MAIN => {
    # some commented line
  }
}
```

### Expected Behavior:

Similarly, the cop should flag the duplicate key "PAS::REST::MAIN" as an offense in the above code snippet. However, it fails to do so.

--------

### Environment:

robocop version: "1.52.1"
